src/App.tsx

```diff
@@ -1,24 +1,166 @@
-/* eslint-disable max-len */
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import { UserWarning } from './UserWarning';
+import { Todo } from './types/Todo';
+import { addTodo, deleteTodo, getTodos } from './api/todos';
+import { TodoForm } from './components/TodoForm';
+import { Status } from './types/Status';
+import { Footer } from './components/Footer';
+import { TodoList } from './components/TodoList';
+import { ErrorNotification } from './components/Error';
+import { Error } from './types/Error';
 
-const USER_ID = 0;
+const USER_ID = 11126;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[] | null>([]);
+  const [loading, setLoading] = useState(false);
+  const [errorNotification, setErrorNotification] = useState('');
+  const [status, setStatus] = useState(Status.All);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [removingId, setRemovingId] = useState<number | null>(null);
+
+  const count = useMemo(() => todos?.reduce(
+    (total, todo) => (todo.completed ? total : total + 1),
+    0,
+  ), [todos]);
+
+  const showErrorNotification = (notification: string): void => {
+    setErrorNotification(notification);
+
+    setInterval(() => {
+      setErrorNotification('');
+    }, 3000);
+  };
```

Comment:
1. Move this logic to Error component
2. setInterval using incorrect here, because this callback function will be triggered every 3 seconds, use setTimeout instead
3. you need to delete the timer if a new error occurs